core/src/ru/spbau/blackout/entities/GameObject.java

```diff
@@ -125,70 +162,73 @@ public final float getHeight() {
         // physics
         public float rotation = DEFAULT_ROTATION;
         public float height = DEFAULT_HEIGHT;
-        private final FixtureDef fixtureDef = new FixtureDef();
-        protected final BodyDef bodyDef = new BodyDef();
+
+        private float density = DEFAULT_DENSITY;
+        /**
+         * As far as Shape itself isn't serializable,
+         * supplier will be sent instead.
+         */
+        private Creator<Shape> shapeCreator;
+        private final Vector2 position = new Vector2();
+
 
         // appearance:
         public String modelPath;
 
-        public Definition(String modelPath, Shape shape, float initialX, float initialY) {
+        /**
+         * ShapeCreator must be serializable.
+         */
+        public Definition(String modelPath, Creator<Shape> shapeCreator,
+                          float initialX, float initialY)
+        {
             this.modelPath = modelPath;
-
-            // setup fixture
-            fixtureDef.shape = shape;
-            fixtureDef.density = DEFAULT_DENSITY;
-            fixtureDef.friction = 0;
-            fixtureDef.restitution = 0;
-
-            // setup body
-            bodyDef.position.set(initialX, initialY);
-            bodyDef.type = getBodyType();
+            this.shapeCreator = shapeCreator;
+            this.position.set(initialX, initialY);
         }
 
         public void setDensity(float density) {
-            fixtureDef.density = density;
+            this.density = density;
         }
 
-        public void setFriction(float friction) {
-            fixtureDef.friction = friction;
+        public void setShapeCreator(Creator<Shape> shapeCreator) {
+            this.shapeCreator = shapeCreator;
         }
 
         public float getDensity() {
-            return fixtureDef.density;
+            return this.density;
         }
 
-        public float getFriction() {
-            return fixtureDef.friction;
+        public Creator<Shape> getShapeCreator() {
+            return this.shapeCreator;
         }
 
         public void setHeight(float height) {
             this.height = height;
         }
 
         public float getHeight() {
-            return height;
+            return this.height;
         }
 
         public Vector2 getPosition() {
-            return bodyDef.position;
+            return this.position;
```

Comment:
It is arguable, isn't it? I didn't found any evidences about which style is preferred in community in general. I prefer this one as well as my Java teacher because it looks clearer and safer for me. Is that style (without extra "this") required for the proect?